File: SlSdkDemo/SdkContract/src/main/java/com/bmtc/sdk/contract/fragment/FundsRateFragment.java
package com.bmtc.sdk.contract.fragment;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;


import com.bmtc.sdk.contract.R;
import com.bmtc.sdk.contract.adapter.FundsRateAdapter;
import com.bmtc.sdk.contract.base.BaseFragment;
import com.bmtc.sdk.contract.utils.ToastUtil;
import com.contract.sdk.ContractPublicDataAgent;
import com.contract.sdk.data.ContractFundingRate;
import com.contract.sdk.data.ContractOrder;
import com.contract.sdk.impl.IResponse;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by zj on 2018/3/1.
 */

public class FundsRateFragment extends BaseFragment {

    private View m_RootView;

    private List<ContractFundingRate> mTradeList = new ArrayList<>();

    private RecyclerView mRecyclerView;
    private FundsRateAdapter mFundsRateAdapter;
    private int mLastVisibleItem;
    private LinearLayoutManager linearLayoutManager;

    private int mContractId;

    public void setContractId(int contractId) {
        mContractId = contractId;
    }
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        m_RootView = inflater.inflate(R.layout.sl_fragment_market_others, null);

        mRecyclerView = m_RootView.findViewById(R.id.rv_list);
        linearLayoutManager = new LinearLayoutManager(getActivity());
        linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
        mRecyclerView.setLayoutManager(linearLayoutManager);
        mRecyclerView.setItemAnimator(new DefaultItemAnimator());

        if (mFundsRateAdapter == null) {
            mFundsRateAdapter = new FundsRateAdapter(getActivity());
            mFundsRateAdapter.setData(mTradeList);
            mRecyclerView.setAdapter(mFundsRateAdapter);
        } else {
            mRecyclerView.setAdapter(mFundsRateAdapter);
        }

        updateData();
        return m_RootView;
    }

    @Override
    public void onResume() {
        super.onResume();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

    public void updateData() {
        if (m_RootView == null) {
            return;
        }

        ContractPublicDataAgent.INSTANCE.loadFundingRate(mContractId, new IResponse<List<ContractFundingRate>>() {
            @Override
            public void onSuccess(@NotNull List<ContractFundingRate> data) {
                if (data != null) {
                    mTradeList.clear();
                    mTradeList.addAll(data);

                    if (mFundsRateAdapter == null) {
                        mFundsRateAdapter = new FundsRateAdapter(getActivity());
                    }

                    mFundsRateAdapter.setData(mTradeList);
                    mFundsRateAdapter.notifyDataSetChanged();
                }
            }

            @Override
            public void onFail(@NotNull String code, @NotNull String msg) {
                ToastUtil.shortToast(getContext(), msg);
            }
        });

    }
}

File: SlSdkDemo/src/main/java/com/yjkj/chainup/contract/fragment/detail/FundsRateFragment.kt
package com.yjkj.chainup.contract.fragment.detail

import android.support.v7.widget.LinearLayoutManager
import android.text.TextUtils
import android.view.LayoutInflater
import android.widget.TextView
import com.contract.sdk.ContractPublicDataAgent
import com.contract.sdk.ContractSDKAgent
import com.contract.sdk.data.Contract
import com.contract.sdk.data.ContractFundingRate
import com.contract.sdk.impl.IResponse
import com.yjkj.chainup.R
import com.yjkj.chainup.base.NBaseFragment
import com.yjkj.chainup.contract.adapter.FundsRateAdapter
import com.yjkj.chainup.contract.utils.onLineText
import com.yjkj.chainup.new_version.view.EmptyForAdapterView
import com.yjkj.chainup.util.ToastUtils
import kotlinx.android.synthetic.main.sl_fragment_cotract_detail_other_recycler.*

/**
 *资金费率
 */
class FundsRateFragment : NBaseFragment(){
    override fun setContentView(): Int {
        return R.layout.sl_fragment_cotract_detail_other_recycler
    }

    var contractId  = 0
    var contract: Contract? = null
    var mList = ArrayList<ContractFundingRate>()
    var adapter: FundsRateAdapter?= null

    override fun loadData() {
        super.loadData()
        contractId = activity?.intent?.getIntExtra("contractId", 0)!!
        contract = ContractPublicDataAgent.getContract(contractId)

        loadDataFromNet()
    }


    override fun initView() {
        recycler_fund_layout.layoutManager = LinearLayoutManager(mActivity)
        adapter = FundsRateAdapter(context!!,mList)
        adapter?.bindToRecyclerView(recycler_fund_layout)
        adapter?.emptyView = EmptyForAdapterView(context?:return)
        recycler_fund_layout.adapter = adapter

        adapter?.settlementInterval = contract?.settlement_interval

        initHeaderView()
    }

    private fun initHeaderView() {
        val headerView = LayoutInflater.from(mActivity).inflate(R.layout.sl_item_header_layout,recycler_fund_layout,false)
        adapter?.addHeaderView(headerView)
        headerView.findViewById<TextView>(R.id.tv_header_left).onLineText("kline_text_dealTime")
        headerView.findViewById<TextView>(R.id.tv_header_right).onLineText("sl_str_funds_rate")
        headerView.findViewById<TextView>(R.id.tv_header_center).onLineText("sl_str_funds_rate_interval")
    }

    private fun loadDataFromNet() {
        ContractPublicDataAgent.loadFundingRate(contractId,object:IResponse<MutableList<ContractFundingRate>>(){
            override fun onSuccess(data: MutableList<ContractFundingRate>) {
                if (data != null) {
                    mList.addAll(data)
                    adapter?.notifyDataSetChanged()
                }
            }

            override fun onFail(code: String, msg: String) {
                ToastUtils.showToast(ContractSDKAgent.context, msg)
            }

        })
    }

}